### PR TITLE
Release 2.10.4

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Comment by tisonkun** [comment]
@liangyepianzhou Can you share the link to vote result or announcement email so that we're sure to merge after released?

**Comment by liangyepianzhou** [comment]
The [vote](https://mail.google.com/mail/u/0/#search/2.10./QgrcJHsBpWZlJHTZLnvvSnGdgJtXFjvHjkL) is completed. We can announce it after this is merged.

File: release-notes/versioned/client-java-2.10.4.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+* [fix][client] Prevent DNS reverse lookup when physical address is an IP address (#19028)
+* [fix][client] Set fields earlier for correct ClientCnx initialization (#19327)
+* [improve][client] Change the get lastMessageId to debug level (#18421)
+* [fix][client] Fix async completion in ConsumerImpl#processPossibleToDLQ (#19392)
+* [fix][client] Broker address resolution wrong if connect through a multi-dns names proxy (#19597)
+* [fix][client] Fix authentication not update after changing the serviceUrl (#19510)
+* [fix][client] Set authentication when using loadConf in client and admin client (#18358)
+* [fix] [client] fix memory leak if enabled pooled messages (#19585)
+* [fix][client][branch-2.10]Return local thread for the newThread (#19779)
+
```
Comment (tisonkun) [inline release-notes/versioned/client-java-2.10.4.md:17]: Let's trim whitespaces.

File: release-notes/versioned/pulsar-2.10.4.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+* [refactor][fn] Use AuthorizationServer more in Function Worker API (#19975)
+
+
+
+### Tiered Storage
+
+
+
+### Pulsar SQL
+
```
Comment (tisonkun) [inline release-notes/versioned/pulsar-2.10.4.md:103]: Ditto. Perhaps drop empty entries.

Comment (tisonkun) [reply release-notes/versioned/pulsar-2.10.4.md:103]: Ditto below

File: release-notes/versioned/pulsar-2.10.4.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+### CLI
+[fix] [cli] Fix Broker crashed by too much memory usage of pulsar tools (#20031)
```
Comment (tisonkun) [inline release-notes/versioned/pulsar-2.10.4.md:106]:
```suggestion
### CLI

* [fix] [cli] Fix Broker crashed by too much memory usage of pulsar tools (#20031)
```

**Review by tisonkun — APPROVED** [review]
Thank you. Some minor comments inline - not a blocker.

**Comment by tisonkun** [comment]
Seems build fails and no related to this patch. I'll investigate it later.

**Comment by tisonkun** [comment]
Weird. I cannot rebase locally:

```
$ gh pr checkout 518
fatal: cannot set up tracking information; starting point 'origin/doc_release_2.10.4' is not a branch
failed to run git: exit status 128
```

Maybe you can rebase @liangyepianzhou

**Comment by liangyepianzhou** [comment]
@tisonkun I have rebased it to the main branch.

**Pull request merged**